Assets/Scripts/BuiltSceneTestScripts/AuthenticationExpirationTestManager.cs

```diff
@@ -57,20 +59,75 @@ public enum AuthenticationProviderToTest
         public Text OriginalToken;
         public Text CurrentToken;
 
-        private Coroutine Probe { get; set; }
+        private Coroutine RunningTestCoroutine { get; set; }
 
+
+        /// <summary>
+        /// <see cref="SetAuthenticationProviderAndLogin(AuthenticationProviderToTest)"/>
+        /// </summary>
+        /// <param name="toSet">
+        /// Integer to cast into a <see cref="AuthenticationProviderToTest"/> argument.
+        /// Needed for the dropdown to function.
+        /// </param>
         public void SetAuthenticationProviderAndLogin(int toSet)
         {
             SetAuthenticationProviderAndLogin((AuthenticationProviderToTest)toSet);
         }
 
+        /// <summary>
+        /// Begins a test of the Authentication token reacquisition.
+        /// This test validates that the various providers correctly have their
+        /// refresh tokens and authentication handled, such that users of the plugin
+        /// will have the login refresh itself correctly when it expires.
+        /// It performs this test by:
+        /// - Logging in using the provider supplied
+        /// - Performs a Connect login
+        /// - Periodically performs an action in EOS
+        ///   If the action fails, the authentication must have failed to refresh.
+        ///   Therefore the test will end then, as a failure.
+        /// - Listens for the authentication provider's token to change.
+        ///   Displays changed tokens, which are generally a proof of this system working.
+        /// The test never finishes 'successfully', so it'll have to declare a success if it doesn't
+        /// fail within some arbitrary time.
+        /// </summary>
+        /// <param name="toSet">The provider to test against.</param>
         public void SetAuthenticationProviderAndLogin(AuthenticationProviderToTest toSet)
         {
+            if (RunningTestCoroutine != null)
+            {
+                Debug.Log($"{nameof(AuthenticationExpirationTestManager)} ({nameof(SetAuthenticationProviderAndLogin)}): Test is already running, cannot run multiple tests in the same instance.");
+                return;
+            }
+
             RunningTest = toSet;
-            StartProbe();
+            StartAuthenticationExpirationProbe();
```

Comment:
In an earlier change within this file, you moved away from using the word `Probe`, is this function's rename consistent with that change?

Reply:
There's now two separate tests, so I renamed this one to be more clear which test this applied to. `StartConnectTest` and `SetAuthenticationProviderAndLogin` are separate tests, and use separate inner functions.